spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/mqtt/MqttHandler.java

```diff
@@ -125,10 +131,13 @@ public void call(Boolean authenticated) {
                         MqttConnectOptions options = new MqttConnectOptions();
                         options.setCleanSession(true);
                         options.setUserName(accessToken);
-                        options.setPassword("anypass".toCharArray());
-                        options.setSocketFactory(
-                            new SCSocketFactory(context.getResources().openRawResource(R.raw.ca))
-                        );
+
+                        if (isSecure) {
+                            options.setPassword("anypass".toCharArray());
```

Comment:
You still need to send a password for certain MQTT brokers even when connecting over plain TCP, so move this one line above and we should be ok.